mimic/test/test_core.py

```diff
@@ -55,3 +58,40 @@ def test_from_plugin_includes_all_plugins(self):
             len(plugin_apis),
             len(list(core.entries_for_tenant('any_tenant', {},
                                              'http://mimic'))))
+
+    def test_load_domain_plugin_includes_all_domain_plugins(self):
+        """
+        Using the :func:`MimicRoot.fromPlugin` creator for a
+        :class:`MimicCore`, domains mocks implementing `class`:`IAPIDomainMock`
+        are included.
+        """
+        # build a fake plugins package with a domain plugin
+        self.root = FilePath(self.mktemp())
+        self.root.createDirectory()
+        self.package = self.root.child('fakeplugins')
+        self.package.createDirectory()
+        init = b"""from fakeplugins._domain import domain_plugin
+__all__ = [domain_plugin]
+"""
+        self.package.child('__init__.py').setContent(init)
+        plugin = b"""from mimic.test.dummy import ExampleDomainAPI
+domain_plugin = ExampleDomainAPI()
+__all__ = [domain_plugin]
+"""
+        self.package.child('_domain.py').setContent(plugin)
+        sys.path.append(self.root.path)
+
+        # monkey patch mimic.plugins package with fakeplugins
+        import fakeplugins
+        import mimic.plugins
+        mimic.plugins = fakeplugins
+
```

Comment:
The important thing here is to put `mimic.plugins` *back* by using [`addCleanup`](https://docs.python.org/2.7/library/unittest.html#unittest.TestCase.addCleanup).  As it is, this corrupts state for all subsequent tests in the process.